PowerEditor/src/WinControls/shortcut/shortcut.cpp

```diff
@@ -1074,12 +1158,28 @@ intptr_t CALLBACK ScintillaKeyMap::run_dlgProc(UINT Message, WPARAM wParam, LPAR
 		{
 			NppDarkMode::autoSubclassAndThemeChildControls(_hSelf);
 
+			mapOemVirtualKeys();
+
 			::SetDlgItemText(_hSelf, IDC_NAME_EDIT, string2wstring(_name, CP_UTF8).c_str());
 			_keyCombo = _keyCombos[0];
 
+			int iPullDown = 0;
 			for (size_t i = 0 ; i < nbKeys ; ++i)
 			{
-				::SendDlgItemMessage(_hSelf, IDC_KEY_COMBO, CB_ADDSTRING, 0, reinterpret_cast<LPARAM>(string2wstring(namedKeyArray[i].name, CP_UTF8).c_str()));
+				const char* nameStr = namedKeyArray[i].name;
+				
+				if (oemVirtualKeyMap.find(namedKeyArray[i].id) != oemVirtualKeyMap.end()) 
+				{
+					nameStr = oemVirtualKeyMap[namedKeyArray[i].id];
+				} 
+				
+				// only add a key to the IDC_KEY_COMBO list if the string is not empty
+				if (nameStr[0])
+				{
+					::SendDlgItemMessage(_hSelf, IDC_KEY_COMBO, CB_ADDSTRING, 0, reinterpret_cast<LPARAM>(string2wstring(nameStr, CP_UTF8).c_str()));
+					oemVirtualKeyUsed[iPullDown] = i;
+					++iPullDown;

```

Comment:
Also we can use the std::vector instead of array for not using the counter.

please use:
```cpp
oemVkUsedIDs.push_back(namedKeyArray[i].id);
```

instead of

```cpp
::SendDlgItemMessage(_hSelf, IDC_KEY_COMBO, CB_ADDSTRING, 0, reinterpret_cast<LPARAM>(string2wstring(nameStr, CP_UTF8).c_str()));
oemVirtualKeyUsed[iPullDown] = i;
++iPullDown;
```